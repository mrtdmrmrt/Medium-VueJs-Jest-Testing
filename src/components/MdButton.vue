<template>
    <button class="md-button" :class="[`md-button--${customClass}`,`md-button--${size}`]">
        <slot />
    </button>   
</template>
<script>
export default {
    props:{
        customClass:{
            type: String,
            default: 'default',
            validator(customClass) {
                return ['default', 'primary','success','warning','danger'].indexOf(customClass) > -1;
            },
        },
        size:{
            type: String,
            default: 'medium',
            validator(size) {
                return ['small', 'medium', 'large'].indexOf(size) > -1;
            },
        }
    }
}
</script>
<style lang="scss">
    .md-button{
        outline: none;
        cursor: pointer;
        display: relative;
        font-weight: 400;
        text-align: center;
        white-space: nowrap;
        vertical-align: middle;
        user-select: none;
        border: 1px solid transparent;
        padding: .375rem .75rem;
        font-size: 1rem;
        line-height: 1.5;
        border-radius: .25rem;
        &--default{
            border-color: darkgrey;
            background-color: white;
            color: darkgrey;
        }
        &--primary{
            color: #fff;
            background-color: #007bff;
            border-color: #007bff;
        }
        &--danger{
            color: #fff;
            background-color: #dc3545;
            border-color: #dc3545;
        }
        &--warning{
            color: #212529;
            background-color: #ffc107;
            border-color: #ffc107;
        }
        &--success{
            color: #fff;
            background-color: #28a745;
            border-color: #28a745;
        }
        &--small {
            font-size: 0.75rem;
        }
        &--medium {
          font-size: 1rem;
        }
        &--large {
          font-size: 1.5rem;
        }
    }
    
</style>