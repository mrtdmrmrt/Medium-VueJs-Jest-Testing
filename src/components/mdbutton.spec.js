import {mount} from '@vue/test-utils'
import MdButton from './MdButton.vue'

describe('MdButton.vue', () => {
  it('Bileşenimizin class özelliğini kontrol edilmesi', () => {
    const wrapper = mount(MdButton)
    expect(wrapper.classes()).toContain('md-button');
  })

 const typeClass = ['default', 'primary','success','warning','danger'];
 typeClass.forEach(item =>{
  it('Bileşenimizin customClass: '+item+' özelliğini props ile kontrol edilmesi',() =>{
    const wrapper = mount(MdButton, {
      propsData: {
        customClass: item
      },
    })
    expect(wrapper.classes()).toContain('md-button--'+item);
   })
 })
 const typeSize = ['small', 'medium', 'large'];
 typeSize.forEach(item =>{
  it('Bileşenimizin size: '+item+' özelliğini props ile kontrol edilmesi',() =>{
    const wrapper = mount(MdButton, {
      propsData: {
        size: item
      },
    })
    expect(wrapper.classes()).toContain('md-button--'+item);
   })
 })
 it('Snapshot özelliği', () => {
  const wrapper = mount(MdButton)
  expect(wrapper.html()).toMatchSnapshot()
})
})

