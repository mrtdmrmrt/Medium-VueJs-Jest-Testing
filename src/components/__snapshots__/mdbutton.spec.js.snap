// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MdButton.vue Snapshot özelliği 1`] = `<button class="md-button md-button--default md-button--medium"></button>`;
